From fa17153ebc83cbc0ae7a1d7430d9509db0c6e0d6 Mon Sep 17 00:00:00 2001
From: Shawn Hoffman <godisgovernment@gmail.com>
Date: Tue, 10 May 2022 16:39:12 -0700
Subject: [PATCH] fmt: use make_format_args instead of make_args_checked
 make_args_checked is deprecated see https://github.com/fmtlib/fmt/pull/2760
 and the linked comment

---
 Source/Core/Common/Logging/Log.h | 3 +--
 Source/Core/Common/MsgHandler.h  | 9 ++-------
 2 files changed, 3 insertions(+), 9 deletions(-)

diff --git a/Source/Core/Common/Logging/Log.h b/Source/Core/Common/Logging/Log.h
index 092a9b4dc6b3..98d420bda2ac 100644
--- a/Source/Core/Common/Logging/Log.h
+++ b/Source/Core/Common/Logging/Log.h
@@ -94,8 +94,7 @@ void GenericLogFmt(LogLevel level, LogType type, const char* file, int line, con
   static_assert(NumFields == sizeof...(args),
                 "Unexpected number of replacement fields in format string; did you pass too few or "
                 "too many arguments?");
-  GenericLogFmtImpl(level, type, file, line, format,
-                    fmt::make_args_checked<Args...>(format, args...));
+  GenericLogFmtImpl(level, type, file, line, format, fmt::make_format_args(args...));
 }
 
 void GenericLog(LogLevel level, LogType type, const char* file, int line, const char* fmt, ...)
diff --git a/Source/Core/Common/MsgHandler.h b/Source/Core/Common/MsgHandler.h
index 0e2de793a1d2..750da515b0cb 100644
--- a/Source/Core/Common/MsgHandler.h
+++ b/Source/Core/Common/MsgHandler.h
@@ -43,7 +43,7 @@ bool MsgAlertFmt(bool yes_no, MsgType style, Common::Log::LogType log_type, cons
                 "too many arguments?");
   static_assert(fmt::is_compile_string<S>::value);
   return MsgAlertFmtImpl(yes_no, style, log_type, file, line, format,
-                         fmt::make_args_checked<Args...>(format, args...));
+                         fmt::make_format_args(args...));
 }
 
 template <std::size_t NumFields, bool has_non_positional_args, typename S, typename... Args>
@@ -57,12 +57,7 @@ bool MsgAlertFmtT(bool yes_no, MsgType style, Common::Log::LogType log_type, con
                 "Unexpected number of replacement fields in format string; did you pass too few or "
                 "too many arguments?");
   static_assert(fmt::is_compile_string<S>::value);
-  // It's only possible for us to compile-time check the English-language string.
-  // make_args_checked uses static_asserts to verify that a string is formattable with the given
-  // arguments.  But it can't do that if the string varies at runtime, so we can't check
-  // translations.  Still, verifying that the English string is correct will help ensure that
-  // translations use valid strings.
-  auto arg_list = fmt::make_args_checked<Args...>(format, args...);
+  auto arg_list = fmt::make_format_args(args...);
   return MsgAlertFmtImpl(yes_no, style, log_type, file, line, translated_format, arg_list);
 }
 
